canary/consumer-java/src/main/java/com/amazon/kinesis/video/canary/consumer/CanaryListFragmentWorker.java

```diff
@@ -0,0 +1,114 @@
+import com.amazonaws.services.kinesisvideo.AmazonKinesisVideo;
+import com.amazonaws.services.kinesisvideo.AmazonKinesisVideoArchivedMedia;
+import com.amazonaws.services.kinesisvideo.AmazonKinesisVideoArchivedMediaClient;
+import com.amazonaws.services.kinesisvideo.model.*;
+import lombok.extern.slf4j.Slf4j;
+
+
+/*
+    This is a modified version of the ListFragmentWorker example found here:
+    https://github.com/aws/amazon-kinesis-video-streams-parser-library/blob/0056da083402a96cffcdfd8bca9433dd25eb7470/src/main/java/com/amazonaws/kinesisvideo/parser/examples/ListFragmentWorker.java#L7
+ */
+
+
+/* This worker retrieves all fragments within the specified TimestampRange from a specified Kinesis Video Stream and returns them in a list */
+@Slf4j
+public class CanaryListFragmentWorker implements Callable {
+    private final FragmentSelector fragmentSelector;
+    private final AmazonKinesisVideoArchivedMedia amazonKinesisVideoArchivedMedia;
+    private final long fragmentsPerRequest = 100;
+    private final Regions region;
+    private final AWSCredentialsProvider credentialsProvider;
+    protected final String streamName;
+
+    public CanaryListFragmentWorker(final String streamName,
+                              final AWSCredentialsProvider credentialsProvider, final String endPoint,
+                              final Regions region,
+                              final FragmentSelector fragmentSelector) {
+        this.region = region;
+        this.credentialsProvider = credentialsProvider;
+        this.streamName = streamName;
+        this.fragmentSelector = fragmentSelector;
+
+        amazonKinesisVideoArchivedMedia = AmazonKinesisVideoArchivedMediaClient
+                .builder()
+                .withCredentials(credentialsProvider)
+                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(endPoint, region.getName()))
+                .build();
+    }
+
+    public static CanaryListFragmentWorker create(final String streamName,
+                                            final AWSCredentialsProvider credentialsProvider,
+                                            final Regions region,
+                                            final AmazonKinesisVideo amazonKinesisVideo,
+                                            final FragmentSelector fragmentSelector) {
+        final GetDataEndpointRequest request = new GetDataEndpointRequest()
+                .withAPIName(APIName.LIST_FRAGMENTS).withStreamName(streamName);
+        final String endpoint = amazonKinesisVideo.getDataEndpoint(request).getDataEndpoint();
+
+        return new CanaryListFragmentWorker(
+                streamName, credentialsProvider, endpoint, region, fragmentSelector);
+    }
+
+    @Override
+    public List<CanaryFragment> call() {
+        List<CanaryFragment> fragments = new ArrayList<>();
+        try {
+            log.info(MessageFormat.format("Start ListFragment worker on stream {0}", streamName));
```

Comment:
`log.info()` works with just `{}` and sans `MessageFormat.format`, but using `System.out.println()` requires it be formatted the way it is. This way, the logging technique can readily be changed. I can revert to be consistent with other canary apps if that is preferred.

Reply:
In see. `System.out.println` is not preferred because we want logs to be pushed to cloudwatch at some point. Lets remove the `MessageFormat.format` usage and be consistent.